lib/websocket/transports/engineio/index.js

```diff
@@ -9,13 +9,13 @@ var openSocketsById = {};
 
 var processSession = function(socket) {
   try {
-    var raw= socket.request.headers.cookie.split(';');
-    if (raw.length === 1) {
-      socket.sessionId = (raw[0].split('='))[1];
-    }
-    else {
-      socket.sessionId = (raw[1].split('='))[1];
-    }
+    socket.request.headers.cookie.split(';').some(function(item) {
+        var item_di = item.split('=');
+        if (item_di[0].trim() == 'connect.sid') {
```

Comment:
It's a bit different now that I've fixed #543 , in that the cookieParser secret is injected.  Are there more config variables that should be brought in here ?